internal/handlers/hana/hanadb/query.go

```diff
@@ -14,8 +14,121 @@
 
 package hanadb
 
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"sync"
+
+	"github.com/FerretDB/FerretDB/internal/handlers/sjson"
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/iterator"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+	"github.com/FerretDB/FerretDB/internal/util/observability"
+	"github.com/FerretDB/FerretDB/internal/util/resource"
+)
+
 // QueryParams represents options/parameters used for SQL query/statement.
 type QueryParams struct {
 	DB         string
 	Collection string
 }
+
+// queryIterator implements iterator.Interface to fetch documents from the database.
+type queryIterator struct {
+	ctx       context.Context
+	unmarshal func(b []byte) (*types.Document, error) // defaults to sjson.Unmarshal
+
+	m    sync.Mutex
+	rows *sql.Rows
+
+	token *resource.Token
+}
+
+// QueryDocuments returns an queryIterator to fetch documents for given SQLParams.
+// If the collection doesn't exist, it returns an empty iterator and no error.
+// If an error occurs, it returns nil and that error, possibly wrapped.
+func (hanaPool *Pool) QueryDocuments(ctx context.Context, qp *QueryParams) (types.DocumentsIterator, error) {
+	// Todo: build correct SQL here
+
+	sqlStmt := fmt.Sprintf("SELECT * FROM %q.%q", qp.DB, qp.Collection)
```

Comment:
`*` could return multiple columns, and that could cause `Scan` below to not work. Let's try to specify a single column name there

Reply:
Seems to make no difference, same error 
```go
[msg_find.go:59 hana.(*Handler).MsgFind] [values.go:74 iterator.ConsumeValuesN[...]] [query.go:101 hanadb.(*queryIterator).Next] sql: Scan error on column index 0, name "test": unsupported Scan, storing driver.Value type *protocol.LobOutDescr into type *[]uint8
```
Maybe there is a different variant of Scan or a way to parse the Scan result for only the json?

Reply:
The issue seems to be coming from https://github.com/SAP/go-hdb, so maybe someone there could help